src/Backend/Core/Installer/ModuleInstaller.php

```diff
@@ -451,539 +465,561 @@ protected function insertDashboardWidget($module, $widget)
                 'users_settings',
                 $settings,
                 'user_id = ? AND name = ?',
-                array($settings['user_id'], $settings['name'])
+                [$settings['user_id'], $settings['name']]
             );
         }
     }
 
     /**
+     * @param string $module
+     *
+     * @return int
+     */
+    private function getNextSequenceForModule(string $module): int
+    {
+        // set next sequence number for this module
+        $sequence = (int) $this->getDB()->getVar(
+            'SELECT MAX(sequence) + 1 FROM modules_extras WHERE module = ?',
+            [$module]
+        );
+
+        // this is the first extra for this module: generate new 1000-series
+        if ($sequence > 0) {
+            return $sequence;
+        }
+
+        return (int) $this->getDB()->getVar(
+            'SELECT CEILING(MAX(sequence) / 1000) * 1000 FROM modules_extras'
+        );
+    }
+
+    /**
      * Insert an extra
      *
-     * @param string $module   The module for the extra.
-     * @param ModuleExtraType|string $type     The type, possible values are: homepage, widget, block.
-     * @param string $label    The label for the extra.
-     * @param string $action   The action.
-     * @param string $data     Optional data, will be passed in the extra.
-     * @param bool   $hidden   Is this extra hidden?
-     * @param int    $sequence The sequence for the extra.
+     * @param string $module The module for the extra.
+     * @param ModuleExtraType $type The type, possible values are: homepage, widget, block.
+     * @param string $label The label for the extra.
+     * @param string $action The action.
+     * @param string $data |null data, will be passed in the extra.
```

Comment:
since it does some extra things I kept it here, also for discoverability but I refactored it a bit
018934202